Improve the stability of the tokenizer

mabels opened the pull request:
Due to some oddities with esm.sh, the object compare of the tokens could fail. 
To fix this, make use of cborg provided Type.compare method.

I'm sorry for the package.json changes but they are needed if I want to run the tests locally.

rvagg commented on package.json:
```diff
@@ -166,6 +166,8 @@
     "@ipld/garbage": "^6.0.0",
-    "aegir": "^45.0.1"
+    "aegir": "^45.0.7",
+    "eslint-config-ipfs": "^7.0.6",
+    "eslint-plugin-n": "^17.14.0"
```
aegir should be contributing what it needs for linting, no? does it fail without these?

mabels replied:
it does fails without on my side.

rvagg replied:
try removing these and push again, this changeset works for me locally without them and pushing a new commit might wake CI up, something odd going in GitHub Actions even after a re-start attempt

rvagg replied:
```
$ npm ls --all | grep 'eslint-config-ipfs\|eslint-plugin-n'
│ │ ├─┬ eslint-config-ipfs@7.0.6
│ │ │ ├─┬ eslint-plugin-n@16.6.2
│ │ │ ├── eslint-plugin-n@16.6.2 deduped
│ │ ├── eslint-plugin-no-only-tests@3.3.0
│ │ ├── eslint-plugin-node@11.1.0 deduped
│ ├─┬ eslint-plugin-node@11.1.0
```

this is from a freshly installed tree without them, so they're there already from aegir.

mabels commented:
The failures in the CI i don't understand --- they are all released to the git checkouts --- any ideas.

mabels commented:
that's what happens here if i remove the eslint deps:
```
rm -rf dist node_modules
npm i%

❯ npm i
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated read-pkg-up@11.0.0: Renamed to read-package-up
npm warn deprecated @babel/plugin-proposal-optional-catch-binding@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-catch-binding instead.
npm warn deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
npm warn deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
npm warn deprecated rimraf@2.7.1: Rimraf versions prior to v4 are no longer supported
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
npm warn deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated @babel/plugin-proposal-object-rest-spread@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-spread instead.
npm warn deprecated boolean@3.2.0: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.

added 1431 packages, and audited 1657 packages in 9s

369 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

❯ npm run build

> @ipld/dag-json@10.2.3 build
> aegir build

[09:13:04] tsc [started]
[09:13:05] tsc [completed]
[09:13:05] esbuild [started]
[09:13:05] esbuild [completed]

❯ npm run test

> @ipld/dag-json@10.2.3 test
> npm run lint && aegir test


> @ipld/dag-json@10.2.3 lint
> aegir lint

[09:13:10] eslint [started]
[09:13:11] eslint [failed]
[09:13:11] → Failed to load plugin 'n' declared in 'package.json » eslint-config-ipfs#overrides[0] » ./js.js » eslint-config-standard': Cannot find module 'eslint-plugin-n'
Require stack:
- /Users/menabe/Software/fproof/js-dag-json/__placeholder__.js
Failed to load plugin 'n' declared in 'package.json » eslint-config-ipfs#overrides[0] » ./js.js » eslint-config-standard': Cannot find module 'eslint-plugin-n'
Require stack:
- /Users/xxxxxx/Software/fproof/js-dag-json/__placeholder__.js
❯
```

rvagg commented:
Sorry, if I run the exact same commands I don't get that error. What version of Node.js and npm are you doing this on?

```
$ npm -v
10.9.0
$ node -v
v23.3.0
```

mabels commented:
> Sorry, if I run the exact same commands I don't get that error. What version of Node.js and npm are you doing this on?
> 
> ```
> $ npm -v
> 10.9.0
> $ node -v
> v23.3.0
> ```

$ node -v
v20.15.1
$ npm -v
10.8.2

if i use:
$ node -v
v23.3.0
$ npm -v
10.9.0

and remove the additional deps --- i get the same error as with the other versions.

my .npmrc a almost clean:
engine-strict=true

that's enough investigation for me now --- I will just remove these imports 
but first, we have to wait for a working cborg to implement .equals.
I didn't look into @ipld/dag-cbor if there is also a .equals update needed.

vmx commented:
I've tried it locally on a fresh checkout without the additional deps. It works for me.

```console
$ node -v
v20.13.1
$ npm -v
10.5.2
```

rvagg commented:
https://github.com/ipld/js-dag-json/pull/141
